katportalclient/test/test_client.py

```diff
@@ -1331,6 +1333,50 @@ def test_modify_userlog(self):
             userlog_to_modify['tags'] = 'random nonsense'
             userlog = yield self._portal_client.modify_userlog(userlog_to_modify)
 
+    @gen_test
+    def test_sensor_subarray_lookup(self):
+        """Test sensor subarray lookup is correctly extracted."""
+        lookup_base_url = self._portal_client.sitemap['subarray'] + '/3/sensor-lookup/cbf/device_status/0'
+        sensor_name_filter = 'device_status'
+        expected_sensor_name = 'cbf_3_device_status'
+      
+        self.mock_http_async_client().fetch.side_effect = mock_async_fetcher(
+                valid_response='{"result":"cbf_3_device_status"}',
+                invalid_response=['error'],
+                starts_with=lookup_base_url,
+                contains=sensor_name_filter)
+        sensor = yield self._portal_client.sensor_subarray_lookup('cbf', sensor_name_filter, False)
+        self.assertTrue(sensor == expected_sensor_name)
+
+    @gen_test
+    def test_sensor_subarray_katcp_name_lookup(self):
+        """Test sensor subarray lookup returns the correct katcp name."""
+        lookup_base_url = self._portal_client.sitemap['subarray'] + '/3/sensor-lookup/cbf/device-status/1'
+        sensor_name_filter = 'device-status'
+        expected_sensor_name = 'cbf_3.device-status'
+
+        self.mock_http_async_client().fetch.side_effect = mock_async_fetcher(
+                valid_response='{"result":"cbf_3.device-status"}',
+                invalid_response=['error'],
+                starts_with=lookup_base_url,
+                contains=sensor_name_filter)
+        sensor = yield self._portal_client.sensor_subarray_lookup('cbf', sensor_name_filter, True)
+        self.assertTrue(sensor == expected_sensor_name)
+
+
```

Comment:
There is one extra blank here.  Normally we want 1 blank line between class methods, and 2 blank lines between functions.